sandbox/auth/session/token.go

```diff
@@ -2,16 +2,40 @@ package session
 
 import (
 	"github.com/go-jose/go-jose/v3"
+	"github.com/go-jose/go-jose/v3/jwt"
 	"golang.org/x/oauth2"
 )
 
+// Tokens can generally be ID or access. It's a bit weird to have a struct
+// named "Token" that only contains an ID token. An alternative is to just rename
+// this struct to IDToken.
 type Token struct {
-	oauth2.Token // Embed and oauth2 token
-
+	// Are these fields needed?
```

Comment:
Issuer and Keys are needed to add validation when deserializing the token. I haven't implemented that, but was hoping to add at some point.

Reply:
Issuer should be a claim in the token and keys should be reachable via jwks endpoint no?

Reply:
Yes, but then you need to:
Decode Token without verifying the signature -> Do a network roundtrip using OIDC discovery to figure out the right endpoints -> Do another network roundtrip to download the keys -> Verify signature

Vs:
Decode and verify token with the "cached" issuer and JWKS data immediately (no network). I know there's an edge case where, if the keys have been rotated, you still need to do another network request to get the latest keys; but that can be done only if the cached data fails to verify the token.